filebeat/config/config.go

```diff
@@ -57,11 +57,11 @@ type ProspectorConfig struct {
 }
 
 type HarvesterConfig struct {
+	common.EventMetadata `config:",inline"` // Fields and tags to add to events.
```

Comment:
+1 on approaching this directly with an eventMetadata object. That makes it easy to extend it later with other fields.